File: 01-Assignment-02-02-2026/Using_C/program_33.c
/* Program 33: Implement a basic calculator */
#include <stdio.h>

float addition(float a, float b){
    return a + b;
}

float subtraction(float a, float b){
    return a - b;
}

float multiplication(float a, float b){
    return a * b;
}

float division(float a, float b){
    return a / b;
}

void GetInfo(int *a, int *b){
    printf("Enter Number 1 : ");
    scanf("%d", a);
    printf("Enter Number 2 : ");
    scanf("%d", b);
}

int main() {
    int a, b;
    char choice;

    printf("Welcome to Basic Calculator\n");

    while (1) {
        printf("\nEnter Your Choice:\n");
        printf("1. Addition -> a\n");
        printf("2. Subtraction -> s\n");
        printf("3. Multiplication -> m\n");
        printf("4. Division -> d\n");
        printf("5. Exit -> e\n");
        printf("Choice: ");

        scanf(" %c", &choice);   // space before %c is IMPORTANT

        switch (choice) {
            case 'a':
                GetInfo(&a, &b);
                printf("Addition of %d and %d is %.2f\n", a, b, addition(a, b));
                break;

            case 's':
                GetInfo(&a, &b);
                printf("Subtraction of %d and %d is %.2f\n", a, b, subtraction(a, b));
                break;

            case 'm':
                GetInfo(&a, &b);
                printf("Multiplication of %d and %d is %.2f\n", a, b, multiplication(a, b));
                break;

            case 'd':
                GetInfo(&a, &b);
                if (b == 0) {
                    printf("Error: Denominator cannot be zero!\n");
                } else {
                    printf("Division of %d and %d is %.2f\n", a, b, division(a, b));
                }
                break;

            case 'e':
                printf("Thanks For Using Us :)\n");
                return 0;

            default:
                printf("Invalid Choice!\n");
        }
    }
}

File: 01-Assignment-02-02-2026/Using_Java/Program33.java
/*
Program 33
Java program to implement a calculator for basic operations
*/

import java.util.Scanner;

public class Program33 {

    public static void main(String[] args) {
        System.out.println("Welcome to Mini Calculator");
        Scanner sc = new Scanner(System.in);
        calculator cal = new calculator();
        while (true) {
            System.out.print(
                    "Choose form these...\n1->adddition\n2->subtraction\n3->division\n4->multiplication\n5->modulus\n6->exit\nchoice ? : ");
            int choice = sc.nextInt();

            if (choice == 6) {
                System.out.println("Bye");
                break;
            } else {
                System.out.print("Enter number a : ");
                double a = sc.nextDouble();
                System.out.print("Enter number b : ");
                double b = sc.nextDouble();
                if (choice == 1) {
                    System.out.println("Result : " + cal.addition(a, b));
                } else if (choice == 2) {
                    System.out.println("Result : " + cal.subtraction(a, b));
                } else if (choice == 3) {
                    System.out.println("Result : " + cal.division(a, b));
                } else if (choice == 4) {
                    System.out.println("Result : " + cal.multiplication(a, b));
                } else if (choice == 5) {
                    System.out.println("Result : " + cal.modulus(a, b));
                } else {
                    System.out.println("Invalid Choice");
                }
            }
        }
        sc.close();
    }

}

class calculator {

    public double addition(double a, double b) {
        return a + b;
    }

    public double subtraction(double a, double b) {
        return a - b;
    }

    public double division(double a, double b) {
        if (b == 0)
            throw new RuntimeException("Denominator cant be zero");
        return a / b;
    }

    public double multiplication(double a, double b) {
        return a * b;
    }

    public double modulus(double a, double b) {
        return a % b;
    }
}
